src/ring_theory/witt_vector/truncated.lean

```diff
@@ -240,3 +238,221 @@ instance : comm_ring (truncated_witt_vector p n R) :=
   (truncate_fun_neg n)
 
 end truncated_witt_vector
+
+namespace witt_vector
+open truncated_witt_vector
+
+variables (n)
+variable [comm_ring R]
+include hp
+
+/-- `truncate n` is a ring homomorphism that truncates `x` to its first `n` entries
+to obtain a `truncated_witt_vector`, which has the same base `p` as `x`. -/
+def truncate : 𝕎 R →+* truncated_witt_vector p n R :=
+{ to_fun := truncate_fun n,
+  map_zero' := truncate_fun_zero p n R,
+  map_add' := truncate_fun_add n,
+  map_one' := truncate_fun_one p n R,
+  map_mul' := truncate_fun_mul n }
+
+variables (p n R)
+
+lemma truncate_surjective : surjective (truncate n : 𝕎 R → truncated_witt_vector p n R) :=
+truncate_fun_surjective p n R
+
+variables {p n R}
+
+@[simp] lemma coeff_truncate (x : 𝕎 R) (i : fin n) :
+  (truncate n x).coeff i = x.coeff i :=
+coeff_truncate_fun _ _
+
+variables (n)
+
+lemma mem_ker_truncate (x : 𝕎 R) :
+  x ∈ (@truncate p _ n R _).ker ↔ ∀ i < n, x.coeff i = 0 :=
+begin
+  simp only [ring_hom.mem_ker, truncate, truncate_fun, ring_hom.coe_mk,
+    truncated_witt_vector.ext_iff, truncated_witt_vector.coeff_mk, coeff_zero],
+  exact subtype.forall
+end
+
+variables (p)
+
+@[simp] lemma truncate_mk (f : ℕ → R) :
+  truncate n (mk p f) = truncated_witt_vector.mk _ (λ k, f k) :=
+begin
+  ext i,
+  rw [coeff_truncate, coeff_mk, truncated_witt_vector.coeff_mk],
+end
+
+end witt_vector
+
+namespace truncated_witt_vector
+
+variable [comm_ring R]
+include hp
+
+/--
+A ring homomorphism that truncates a truncated Witt vector of length `m` to
+a truncated Witt vector of length `n`, for `n ≤ m`.
```

Comment:
It would be interesting, and probably easy (although admittedly not crucial) to introduce the shift map on truncated Witt vectors and its relation to Verschiebung as in https://ncatlab.org/nlab/show/ring+of+Witt+vectors#the_shift_map

Reply:
I would prefer to do this in another PR.

Reply:
Sure